### PR TITLE
misc: Docstring updates

### PR DESCRIPTION
**Pull request by ZoeLeibowitz**
Started updating the docstrings for the website. Just want to check I am doing it correctly before I finish them...

### CODE REVIEW

**Comment by mloubout** [comment]
Looks all good to me.

FYI, if you clone the website repo, you can render it locally to check if it looks like what you want

File: devito/arch/compiler.py [hunk 1]
```diff
@@ -163,9 +163,9 @@ def __init__(self):
     suffix : str, optional
         The JIT compiler version to be used. For example, assuming ``CC=gcc`` and
         ``suffix='4.9'``, the ``gcc-4.9`` will be used as JIT compiler.
-    cpp : bool, optional
+    cpp : bool, optional, default=False
```
Comment (FabioLuporini) [inline devito/arch/compiler.py:166]: is this format canonical in numpydoc ?

We never use `, defaulat=...` on the first line

File: devito/types/sparse.py [hunk 2]
```diff
@@ -1148,19 +1148,19 @@ class PrecomputedSparseTimeFunction(AbstractSparseTimeFunction,
         So for `r=6`, we will store 18 coefficients per sparse point (instead of
         potentially 216).  Must be a three-dimensional array of shape
         `(npoint, grid.ndim, r)`.
-    space_order : int, optional
+    space_order : int, optional, default=0
         Discretisation order for space derivatives. Defaults to 0.
```
Comment (FabioLuporini) [inline devito/types/sparse.py:1152]: Now all the "Defaults to ..." in the subsequent lines become redundant so they should maybe be dropped. There are quite a few

**Review by FabioLuporini — CHANGES_REQUESTED** [review]

File: devito/arch/compiler.py (hunk 1 above)
Comment (mloubout) [reply devito/arch/compiler.py:166]: Yeah it is

https://numpydoc.readthedocs.io/en/latest/format.html#parameters

**Comment by mloubout** [comment]
@ZoeLeibowitz do you plan on making more change or should this be merged?

**Comment by mloubout** [comment]
Ok thanks, let us know when it's ready

**Comment by ZoeLeibowitz** [comment]
> Ok thanks, let us know when it's ready

Hi, since I am not 100% sure, I will not make more changes

**Review by FabioLuporini — COMMENTED** [review]
@ZoeLeibowitz could you rebase and push again? in absence of code conflicts, I'd like to merge it

You will have to update the notebooks as well, or some tests will fail.

Have you ever run py.test on our jupyter notebooks locally supplying the `--nbval` argument?

**Comment by ZoeLeibowitz** [comment]
> ah yes there are now some conflicts in dense.py

Ah yes, I think it should be fixed now but could you trigger the CI to check please?

**Review by FabioLuporini — APPROVED** [review]
LGTG.

**Comment by FabioLuporini** [comment]
Thanks, landing!

**Pull request merged**